File: 849_maximize_distance_to_closest_person/maxDistToClosest.py
from math import ceil
from typing import List


class Solution:
    def maxDistToClosest(self, seats: List[int]) -> int:
        m = 0
        n = len(seats)

        curr = 0
        start = 0

        for i in range(n):
            if seats[i] == 0:
                curr += 1
                if i == n - 1 or start == 0:
                    m = max(m, curr)
                else:
                    m = max(m, ceil(curr / 2))
            else:
                curr = 0
                start = i + 1


        return m

File: 849_maximize_distance_to_closest_person/maxDistToClosest.ts
function maxDistToClosest(seats: number[]): number {
  const n = seats.length;

  let max = 0;
  let count = 0;
  let i = 0;

  if (seats[0] === 0) {
    while (seats[i] === 0) {
      count++;
      max = Math.max(max, count);
      i++;
    }
  }

  for (; i < n; i++) {
    if (seats[i] === 0) {
      count++;
      if (i === n - 1) {
        max = Math.max(max, count);
      } else {
        max = Math.max(max, Math.ceil(count / 2));
      }
    } else {
      count = 0;
    }
  }

  return max;
}
